### PR TITLE
Generalised scheme for fixing SCF convergence problem due to degeneracy in MO energies

### PR DESCRIPTION
**Pull request by Pulokesh**
1. In x0.scf.in and x1.scf.in, initial geometry is changed to that geometry where degeneracy occurs.
2. In x_to_libra_qe.py, actual scheme is inserted.
3. In main.py, some changes are made in order to tackle degeneracy problem in the very first SCF calculation.

4. In main.py, write_qe_input_first() is used which does not use mol object.

### CODE REVIEW

File: src/extract_qe.py [hunk 1]
```diff
@@ -186,9 +185,211 @@ def qe_extract_gradients(inp_str,  flag):
     # Gradients, in units Ha/Bohr
     return grads
 
+#alp_d,bet_d = check_eig_deg()
+
+def fermi_pop(e):
```
Comment (alexvakimov) [inline src/extract_qe.py]: please document the function

File: src/create_input_qe.py [hunk 2]
```diff
@@ -218,4 +227,45 @@ def write_qe_input(ex_st, label, mol, params):
         
     g.close()
 
+def write_qe_input_first(filename,occ,occ_alp,occ_bet,nspin):
```
Comment (alexvakimov) [inline src/create_input_qe.py]: documentation is needed

File: src/extract_qe.py [hunk 3]
```diff
@@ -186,9 +185,211 @@ def qe_extract_gradients(inp_str,  flag):
...
+    os.system("srun pw.x < %s > %s" % (inp,out))
+    os.system("srun pw_export.x < %s > %s" % (inexp,outexp))
+
+    # Delete scratch directory and unecessary files
+    #os.system("rm *.dat *.wfc* *.igk* *.mix*")
+    #os.system("rm -r *.save") # not sure if we  need to remove this directory
+
+
+
+def robust_cal_extract(filename, ex_st, nel, flag_a, flag_b):
```
Comment (alexvakimov) [inline src/extract_qe.py]: do we still need this function?

**Comment by alexvakimov** [comment]
Looks good, only need to put the proper documentation

File: src/extract_qe.py (hunk 3 above)
Comment (Pulokesh) [inline src/extract_qe.py]: No, we dont need this, I will delete it. There are additional functions in extract_qe.py which are also going to be deleted.

**Pull request merged**